softwerkskammer/lib/mailarchive/importMails.js

```diff
@@ -23,22 +24,21 @@ module.exports = function importMails(file, group, done) {
     return moment();
   }
 
-  function assignMessageId(parsedObject, mailDbObject, done1) {
+  function assignMessageId(parsedObject, mailDbObject, callback) {
     /* eslint new-cap: 0 */
     if (parsedObject.messageId) {
       mailDbObject.id = parsedObject.messageId;
-      done1(null, mailDbObject);
-      return;
+      return callback();
     }
 
     const shasum = crypto.createHash('sha1');
-    const s = fs.ReadStream(file);
+    const s = fs.createReadStream(file); // file may be big -> stream instead of readFile
+
     s.on('data', d => shasum.update(d));
 
     s.on('end', () => {
-      const d = shasum.digest('hex');
-      mailDbObject.id = 'mail-sha1-' + d + '@softwerkskammer.org';
-      done1(null, mailDbObject);
+      mailDbObject.id = 'mail-sha1-' + shasum.digest('hex') + '@softwerkskammer.org';
+      callback();
```

Comment:
Oh, na DAS wäre doch mal was zum Umbauen! Etwa so, dass diese Funktion nur die ID zurückliefert und die dann außen in das Objekt gesetzt wird. So ist das doch absolut gruselig.